### PR TITLE
Fix/client and server are not blocked by exceptions

### PR DESCRIPTION
**Pull request by domef**
The client can set a timeout when sending a message to the server. In this way, if the server raises an exception, the client will not be stopped indefinitely.

### CODE REVIEW

File: shiva/__init__.py [hunk 1]
```diff
@@ -991,8 +1037,12 @@ async def create_and_connect(
         await client.connect()
         return client
 
-    async def send_message(self, message: ShivaMessage) -> ShivaMessage:
+    async def send_message(
+        self, message: ShivaMessage, timeout: float = 0
+    ) -> ShivaMessage:
         await ShivaMessage.send_message_async(self._writer, message)
-        responose_message = await ShivaMessage.receive_message_async(self._reader)
+        responose_message = await ShivaMessage.receive_message_async(
```
Comment (lykius) [inline shiva/__init__.py]: now it's a good time to fix this typo :)

**Review by lykius — CHANGES_REQUESTED** [review]
see comment

**Review by lykius — APPROVED** [review]
Lucertola Giaguaro Topo Minotauro

**Comment by mmarcoo** [comment]
@domef @lykius può essere conclusa?

**Comment by lykius** [comment]
merged

**Pull request merged**